pkg/daemon/daemon.go

```diff
@@ -496,16 +481,21 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 			return err
 		}
 	}
+
+	if utils.NodeHasAnnotation(*dn.node, consts.NodeDrainAnnotation, annoDrainRequired) {
+		glog.Info("nodeStateSyncHandler(): waiting for drain")
+		return nil
+	}
+
 	if reqDrain {
 		if !dn.isNodeDraining() {
 			if !dn.disableDrain {
-				ctx, cancel := context.WithCancel(context.TODO())
-				defer cancel()
+				glog.Infof("nodeStateSyncHandler(): apply 'Drain_Required' label for node")
 
-				glog.Infof("nodeStateSyncHandler(): get drain lock for sriov daemon")
-				done := make(chan bool)
-				go dn.getDrainLock(ctx, done)
-				<-done
+				if err := dn.applyDrainRequired(); err != nil {
```

Comment:
this will enter into a loop as for OCP the isNodeDraining will be true if we pass the pauseMCP function

Reply:
done

Reply:
If `IsOpenshiftCluster() == true`, we never go here and always pauseMCP and drain the node, despite of how many nodes are currently draining. Am I right?

(See [L500](https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/427/files#diff-a53b7b593d3d778e62eaeeafa40088656f9212bfa2c2b7991df15fa78e60b0f0R500))